tests/unit/backend/utils/field-validation/attachment-validation.spec.ts

```diff
@@ -0,0 +1,174 @@
+import { ValidateFieldError } from 'src/app/modules/submission/submission.errors'
+import { ProcessedAttachmentResponse } from 'src/app/modules/submission/submission.types'
+import { validateField } from 'src/app/utils/field-validation/'
+import { IFieldSchema } from 'src/types'
+
+type MakeFieldOptions = {
+  required?: boolean
+}
+
+type MakeResponseOptions = {
+  answer?: string
+  filename?: string
+}
+
+describe('Attachment validation', () => {
+  const makeField = (
+    fieldId: string,
+    size: string,
+    options?: MakeFieldOptions,
+  ) => {
+    const attachment = {
+      _id: fieldId,
+      fieldType: 'attachment',
+      attachmentSize: size,
+      required: true,
+      ...options,
+    }
+    return (attachment as unknown) as IFieldSchema
+  }
+
+  const makeResponse = (
+    fieldId: string,
+    buffer?: Buffer,
+    options?: MakeResponseOptions,
+  ) => {
+    const response = {
+      _id: fieldId,
+      fieldType: 'attachment',
+      answer: 'file.jpg',
+      filename: 'file.jpg',
+      content: buffer,
+      isVisible: true,
+      ...options,
+    }
+    return (response as unknown) as ProcessedAttachmentResponse
+  }
+
+  const formId = '5dd3b0bd3fbe670012fdf23f'
+  const fieldId = '5ad072e3d9a3d4000f2c77c8'
+
+  describe('Required or optional', () => {
```

Comment:
can we also add a test for the valid cases? required with answer, optional with answer